Commit 5c3c353 introduced some error that breaks all samples

# Bug Report

## Describe the Bug

I did a bisection and found that commit 5c3c353 introduced some bug that causes pretty much all samples to throw injection errors. For instance, basic test 02:


When I run
```
./gradlew clean basic:basic-02-health-endpoint:build
java -jar ./basic/basic-02-health-endpoint/build/libs/connector-health.jar
```
The build finishes correctly, but when running the connector, I get an error:
```
INFO 2025-09-18T14:21:56.659189335 Booting EDC runtime
WARNING 2025-09-18T14:21:56.698978068 The runtime is configured as an anonymous participant. DO NOT DO THIS IN PRODUCTION.
SEVERE 2025-09-18T14:21:56.755933908 Error booting runtime: The following problems occurred during dependency injection: ## class org.eclipse.edc.extension.health.HealthEndpointExtension is missing   --> Field "webService" of type [interface org.eclipse.edc.web.spi.WebService]
org.eclipse.edc.boot.system.injection.EdcInjectionException: The following problems occurred during dependency injection:
## class org.eclipse.edc.extension.health.HealthEndpointExtension is missing
  --> Field "webService" of type [interface org.eclipse.edc.web.spi.WebService]
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:99)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:80)
SEVERE 2025-09-18T14:21:56.756978962 Error booting runtime: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following problems occurred during dependency injection: ## class org.eclipse.edc.extension.health.HealthEndpointExtension is missing   --> Field "webService" of type [interface org.eclipse.edc.web.spi.WebService]
org.eclipse.edc.spi.EdcException: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following problems occurred during dependency injection:
## class org.eclipse.edc.extension.health.HealthEndpointExtension is missing
  --> Field "webService" of type [interface org.eclipse.edc.web.spi.WebService]
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.onError(BaseRuntime.java:148)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:99)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:80)
Caused by: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following problems occurred during dependency injection:
## class org.eclipse.edc.extension.health.HealthEndpointExtension is missing
  --> Field "webService" of type [interface org.eclipse.edc.web.spi.WebService]
        ... 2 more
Exception in thread "main" org.eclipse.edc.spi.EdcException: org.eclipse.edc.spi.EdcException: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following problems occurred during dependency injection:
## class org.eclipse.edc.extension.health.HealthEndpointExtension is missing
  --> Field "webService" of type [interface org.eclipse.edc.web.spi.WebService]
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.onError(BaseRuntime.java:148)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:117)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:80)
Caused by: org.eclipse.edc.spi.EdcException: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following problems occurred during dependency injection:
## class org.eclipse.edc.extension.health.HealthEndpointExtension is missing
  --> Field "webService" of type [interface org.eclipse.edc.web.spi.WebService]
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.onError(BaseRuntime.java:148)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:99)
        ... 1 more
Caused by: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following problems occurred during dependency injection:
## class org.eclipse.edc.extension.health.HealthEndpointExtension is missing
  --> Field "webService" of type [interface org.eclipse.edc.web.spi.WebService]
        ... 2 more
```

### Expected Behavior

Connector starts and stays on.

### Observed Behavior

Error appears instantly.

## Steps to Reproduce

1. Checkout 5c3c353 (at the moment of writing this, main)
2. Build basic-02 sample (or pretty much any other) `./gradlew clean basic:basic-02-health-endpoint:build`
3. Execute the connector `java -jar ./basic/basic-02-health-endpoint/build/libs/connector-health.jar`

## Context Information

- OS: Tried both Windows and WSL

--- a/advanced/advanced-02-custom-runtime/build.gradle.kts
+++ b/advanced/advanced-02-custom-runtime/build.gradle.kts
@@ -37,4 +37,5 @@ tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("custom-runtime.jar")
     dependsOn(distTar, distZip)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/basic/basic-01-basic-connector/README.md
+++ b/basic/basic-01-basic-connector/README.md
@@ -36,7 +36,7 @@ With that we can build and run the connector from the root directory:
 
 ```bash
 ./gradlew clean basic:basic-01-basic-connector:build
-java -jar basic/basic-01-basic-connector/build/libs/basic-connector.jar --log-level=DEBUG
+java -jar basic/basic-01-basic-connector/build/libs/basic-connector.jar
 ```
 
 _Note: the above snippet assumes that you did not alter the build file, i.e. the `shadow` plugin is used and the build
@@ -46,25 +46,13 @@ installation._
 If everything works as intended you should see command-line output similar to this:
 
 ```bash
-INFO 2022-01-13T13:43:57.677973407 Secrets vault not configured. Defaulting to null vault.
-INFO 2022-01-13T13:43:57.680158117 Initialized Null Vault
-INFO 2022-01-13T13:43:57.851181615 Initialized Core Services
-INFO 2022-01-13T13:43:57.852046576 Initialized Schema Registry
-INFO 2022-01-13T13:43:57.853010987 Initialized In-Memory Transfer Process Store
-INFO 2022-01-13T13:43:57.856956651 Initialized Core Transfer
-INFO 2022-01-13T13:43:57.857664924 Initialized In-Memory Asset Index
-INFO 2022-01-13T13:43:57.857957714 Initialized In-Memory Contract Definition Store
-INFO 2022-01-13T13:43:57.860738605 Initialized Core Contract Service
-INFO 2022-01-13T13:43:57.861390422 Initialized In-Memory Contract Negotiation Store
-INFO 2022-01-13T13:43:57.862002044 Started Core Services
-INFO 2022-01-13T13:43:57.862247712 Started Schema Registry
-INFO 2022-01-13T13:43:57.862782289 Started In-Memory Transfer Process Store
-INFO 2022-01-13T13:43:57.8635804 Started Core Transfer
-INFO 2022-01-13T13:43:57.86371948 Started In-Memory Asset Index
-INFO 2022-01-13T13:43:57.863838751 Started In-Memory Contract Definition Store
-INFO 2022-01-13T13:43:57.86497334 Started Core Contract Service
-INFO 2022-01-13T13:43:57.865146132 Started In-Memory Contract Negotiation Store
-INFO 2022-01-13T13:43:57.866073376 edc-e796b518-35f0-4c45-a333-79ca20a6be06 ready
+INFO 2025-09-22T09:28:34.533664939 Booting EDC runtime
+WARNING 2025-09-22T09:28:34.557375883 The runtime is configured as an anonymous participant. DO NOT DO THIS IN PRODUCTION.
+INFO 2025-09-22T09:28:34.6675347 HTTPS enforcement it not enabled, please enable it in a production environment
+WARNING 2025-09-22T09:28:34.774747827 Config value: no setting found for 'edc.hostname', falling back to default value 'localhost'
+WARNING 2025-09-22T09:28:34.88495366 Using the InMemoryVault is not suitable for production scenarios and should be replaced with an actual Vault!
+INFO 2025-09-22T09:28:35.031852619 9 service extensions started
+INFO 2025-09-22T09:28:35.032885401 Runtime e1fb90ce-63a0-4d6b-bcdf-8c998b67f41b ready
 ```
 
 This basic connector - while perfectly fine - does not offer any outward-facing API, nor does it provide any

--- a/basic/basic-01-basic-connector/build.gradle.kts
+++ b/basic/basic-01-basic-connector/build.gradle.kts
@@ -33,4 +33,5 @@ application {
 tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("basic-connector.jar")
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/basic/basic-02-health-endpoint/build.gradle.kts
+++ b/basic/basic-02-health-endpoint/build.gradle.kts
@@ -36,4 +36,5 @@ application {
 tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("connector-health.jar")
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/basic/basic-03-configuration/build.gradle.kts
+++ b/basic/basic-03-configuration/build.gradle.kts
@@ -39,4 +39,5 @@ application {
 tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("filesystem-config-connector.jar")
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/federated-catalog/fc-01-embedded/fc-connector/build.gradle.kts
+++ b/federated-catalog/fc-01-embedded/fc-connector/build.gradle.kts
@@ -45,4 +45,5 @@ tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("fc-connector.jar")
     dependsOn(distTar, distZip)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/federated-catalog/fc-02-standalone/standalone-fc/build.gradle.kts
+++ b/federated-catalog/fc-02-standalone/standalone-fc/build.gradle.kts
@@ -45,4 +45,5 @@ tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("standalone-fc.jar")
     dependsOn(distTar, distZip)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/federated-catalog/fc-03-static-node-directory/embedded-fc-with-node-resolver/build.gradle.kts
+++ b/federated-catalog/fc-03-static-node-directory/embedded-fc-with-node-resolver/build.gradle.kts
@@ -45,4 +45,5 @@ tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("fc-connector-with-node-resolver.jar")
     dependsOn(distTar, distZip)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/federated-catalog/fc-03-static-node-directory/standalone-fc-with-node-resolver/build.gradle.kts
+++ b/federated-catalog/fc-03-static-node-directory/standalone-fc-with-node-resolver/build.gradle.kts
@@ -46,4 +46,5 @@ tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("standalone-fc-with-node-resolver.jar")
     dependsOn(distTar, distZip)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/policy/policy-01-policy-enforcement/policy-enforcement-consumer/build.gradle.kts
+++ b/policy/policy-01-policy-enforcement/policy-enforcement-consumer/build.gradle.kts
@@ -37,4 +37,5 @@ application {
 tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("consumer.jar")
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/policy/policy-01-policy-enforcement/policy-enforcement-provider/build.gradle.kts
+++ b/policy/policy-01-policy-enforcement/policy-enforcement-provider/build.gradle.kts
@@ -40,4 +40,5 @@ application {
 tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("provider.jar")
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/transfer/transfer-00-prerequisites/connector/build.gradle.kts
+++ b/transfer/transfer-00-prerequisites/connector/build.gradle.kts
@@ -59,4 +59,5 @@ tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("connector.jar")
     dependsOn(distTar, distZip)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/transfer/transfer-03-consumer-pull/provider-proxy-data-plane/build.gradle.kts
+++ b/transfer/transfer-03-consumer-pull/provider-proxy-data-plane/build.gradle.kts
@@ -41,4 +41,5 @@ tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("connector.jar")
     dependsOn(distTar, distZip)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/transfer/transfer-04-event-consumer/consumer-with-listener/build.gradle.kts
+++ b/transfer/transfer-04-event-consumer/consumer-with-listener/build.gradle.kts
@@ -62,4 +62,5 @@ tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("connector.jar")
     dependsOn(distTar, distZip)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/transfer/transfer-05-file-transfer-cloud/cloud-transfer-consumer/build.gradle.kts
+++ b/transfer/transfer-05-file-transfer-cloud/cloud-transfer-consumer/build.gradle.kts
@@ -54,4 +54,5 @@ application {
 tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("consumer.jar")
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/transfer/transfer-05-file-transfer-cloud/cloud-transfer-provider/build.gradle.kts
+++ b/transfer/transfer-05-file-transfer-cloud/cloud-transfer-provider/build.gradle.kts
@@ -57,4 +57,5 @@ application {
 tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("provider.jar")
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

--- a/transfer/transfer-06-kafka-broker/kafka-runtime/build.gradle.kts
+++ b/transfer/transfer-06-kafka-broker/kafka-runtime/build.gradle.kts
@@ -36,4 +36,5 @@ application {
 tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
     mergeServiceFiles()
     archiveFileName.set("connector.jar")
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
 }

